src/components/game-board/GameBoard.tsx

```diff
@@ -68,31 +71,37 @@ function GameBoard({ className }: ChildProps): React.JSX.Element{
 
             };
         });
-    }, [setGame]);
+    }, []);
 
     // useEffect to initialize the game
     useEffect(() => {
+
         setGame((prevGame: Game) => ({
             ...prevGame,
             content: prevGame.starter || getOpener(prevGame),
             activePlayer: prevGame.players[0],
             nextPlayer: prevGame.players[1]
         }));
-    }, [getOpener]);
+    }, [showGameDialog, getOpener]);
 
     return (<div className= {className}>
-            <SidePanel className='flex basis-1/3 flex-col justify-center'
-                       game={game}
-                       endGame={setEndGame}
-                       updatePlayerTurn={updatePlayerTurn}>
-            </SidePanel>
-            <StoryBoard className='flex basis-2/3 border-2
+        <TriggerProvider>
+        <SidePanel className='flex basis-1/3 flex-col justify-center'
+                   game={game}
+                   endGame={setEndGame}
+                   updatePlayerTurn={updatePlayerTurn}>
+        </SidePanel>
+        <StoryBoard className='flex basis-2/3 border-2
             max-2xl board-container flex-col p-6
              relative justify-center align-middle items-center'
-                        content={game.content}
-                        activePlayer={game?.activePlayer}
-                        updatePlayerTurn={updatePlayerTurn}>
-            </StoryBoard>
+                    content={game.content}
+                    activePlayer={game?.activePlayer}
+                    updatePlayerTurn={updatePlayerTurn}>
+        </StoryBoard>
+        <StartGameDialog
+            className='bg-gray-800 text-white text-2xl min-w-5 p-6 rounded-lg shadow-xl backdrop:bg-gray-900/50'
```

Comment:
Don't you think some classes should be in the Dialog component?  So it has it own style